src/mlpack/methods/ann/layer/gru.hpp

```diff
@@ -0,0 +1,195 @@
+/**
+ * @file methods/ann/layer/gru.hpp
+ * @author Sumedh Ghaisas
+ * @author Zachary Ng
+ *
+ * Definition of the GRU layer.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_GRU_HPP
+#define MLPACK_METHODS_ANN_LAYER_GRU_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+
+/**
+ * An implementation of a gru network layer using the following algorithm.
+ * 
+ * r_t = sigmoid(W_r x_t + U_r y_{t - 1})
+ * z_t = sigmoid(W_z x_t + U_z y_{t - 1})
+ * h_t =    tanh(W_h x_t + r_t % (U_h y_{t - 1}))
+ * y_t =        (1 - z_t) % y_{t - 1} + z_t % h_t
+ *
+ * For more information, read the following paper:
+ *
+ * @code
+ * @inproceedings{chung2015gated,
+ *    title     = {Gated Feedback Recurrent Neural Networks.},
```

Comment:
```suggestion
 *    title     = {Gated Feedback Recurrent Neural Networks},
```
This may be the most trivial, pedantic comment I have ever left.  (Probably not.  I leave a lot of trivial and pedantic comments :))